src/Compilers/Core/Portable/SourceGeneration/Nodes/InputNode.cs

```diff
@@ -38,7 +38,7 @@ public NodeStateTable<T> UpdateStateTable(DriverStateTable.Builder graphState, N
             var inputItems = _getInput(graphState);
 
             // create a mutable hashset of the new items we can check against
-            HashSet<T> itemsSet = new HashSet<T>(_comparer);
+            HashSet<T> itemsSet = new HashSet<T>();
```

Comment:
Why aren't we using the custom comparer here?

Reply:
That's the bug. Because these are 'input' items (compilations / syntax trees / additional files etc) we control their lifetime. If the user comparer is used they can say that two additional files are different when they are not, that breaks our internal assumptions around comparers that they can only be used to change a modified value back to a cached value.

Basically the only time the comparer should be used is if we think the item has changed, the user has to the ability to override it.